src/test/FuzzerImpl.cpp

```diff
@@ -33,15 +33,62 @@ auto constexpr NUMBER_OF_ASSETS_TO_ISSUE = 4;
 auto constexpr FUZZING_FEE = 1;
 auto constexpr FUZZING_RESERVE = 4;
 auto constexpr INITIAL_TRUST_LINE_LIMIT = 3 * INITIAL_ASSET_DISTRIBUTION;
+auto constexpr DEFAULT_NUM_TRANSACTIONS_TO_RESERVE_FEES_FOR = 10;
+auto constexpr DEFAULT_STROOPS_AVAILABLE_FOR_TEST_ACTIVITY = 32LL;
 
 // must be strictly less than 255
 uint8_t constexpr NUMBER_OF_PREGENERATED_ACCOUNTS = 16;
 
-// As a starting point, twice what we need for setup.
-auto constexpr INITIAL_STROOP_BALANCE =
-    2LL * (2 * FUZZING_RESERVE + (FUZZING_FEE + INITIAL_ASSET_DISTRIBUTION) *
-                                     NUMBER_OF_ASSETS_TO_ISSUE *
-                                     NUMBER_OF_PREGENERATED_ACCOUNTS);
+class AccountParameters
+{
+  public:
+    size_t mNumOffers;
+    int64_t mOfferLiabilities;
+    size_t mNumTransactionsToReserveFeesFor;
+    size_t mNumAssetsToDistribute;
+    int64_t mAvailableForTestActivity;
+
+    AccountParameters()
+        : mNumOffers(0)
+        , mOfferLiabilities(0)
+        , mNumTransactionsToReserveFeesFor(
+              DEFAULT_NUM_TRANSACTIONS_TO_RESERVE_FEES_FOR)
+        , mNumAssetsToDistribute(NUMBER_OF_ASSETS_TO_ISSUE)
+        , mAvailableForTestActivity(DEFAULT_STROOPS_AVAILABLE_FOR_TEST_ACTIVITY)
+    {
+    }
+};
+
+class OfferParameters
+{
+  public:
+    stellar::Asset mBid;
+    stellar::Asset mSell;
+    int mPrivateKey;
+    Price mPrice;
+    int64_t mAmount;
+
+    OfferParameters(stellar::Asset const& bid, stellar::Asset const& sell,
+                    int const& privateKey, Price const& price,
+                    int64_t const amount)
+        : mBid(bid)
+        , mSell(sell)
+        , mPrivateKey(privateKey)
+        , mPrice(price)
+        , mAmount(amount)
+    {
+    }
+};
+
+int64_t
+initialAccountBalance(AccountParameters const& parameters)
+{
+    return (2 + parameters.mNumOffers) * FUZZING_RESERVE +
```

Comment:
> You aren't accounting for the number of trust lines.

Good point.  Added.

> Why does this depend on the number of assets to distribute? Those assets don't have a native balance so that shouldn't matter at all here.

That code derived from the same misunderstanding of liabilities that you pointed out in other comments.  I've removed it.